tests/fermi/test_fermionic.py

```diff
@@ -97,37 +103,286 @@ def test_compact(self, fw, str_rep):
+        (fw3, FermiSentence({fw1: 1.2, fw3: 3j}), FermiSentence({fw1: -1.2, fw3: (1 - 3j)})),
+        (fw1, FermiSentence({fw1: -1.2, fw3: 3j}), FermiSentence({fw1: 2.2, fw3: -3j})),
+    ]
+
+    @pytest.mark.parametrize("w, s, res", WORDS_AND_SENTENCES_SUB)
+    def test_subtract_fermi_words_and_sentences(self, w, s, res):
+        """Test that subtracting a FermiSentence from a FermiWord returns the expected FermiSentence"""
+        diff = w - s
+        # due to rounding, the actual result for floats is
+        # e.g. -0.19999999999999... instead of 0.2, so we round to compare
+        diff_rounded = FermiSentence(
+            {k: round(v, 2) if isinstance(v, float) else v for k, v in diff.items()}
+        )
+
+        assert diff_rounded == res
+
+    WORDS_AND_CONSTANTS_SUBTRACT = [
+        (fw1, 5, FermiSentence({fw1: 1, fw4: -5})),  # int
+        (fw2, 2.8, FermiSentence({fw2: 1, fw4: -2.8})),  # float
+        (fw3, (1 + 3j), FermiSentence({fw3: 1, fw4: -(1 + 3j)})),  # complex
+        (fw1, np.array([5]), FermiSentence({fw1: 1, fw4: -5})),  # numpy array
+        (fw2, pnp.array([2.8]), FermiSentence({fw2: 1, fw4: -2.8})),  # pennylane numpy array
+        (fw4, 2, FermiSentence({fw4: -1})),  # FermiWord is Identity
+    ]
+
+    @pytest.mark.parametrize("w, c, res", WORDS_AND_CONSTANTS_SUBTRACT)
+    def test_subtract_constant_from_fermi_word(self, w, c, res):
+        """Test that subtracting a constant (int, float or complex) from a FermiWord
+        returns the expected FermiSentence"""
+        diff = w - c
+        # due to rounding, the actual result for floats is
+        # e.g. -0.19999999999999... instead of 0.2, so we round to compare
+        diff_rounded = FermiSentence(
+            {k: round(v, 2) if isinstance(v, float) else v for k, v in diff.items()}
+        )
+        assert diff_rounded == res
+
+    CONSTANTS_AND_WORDS_SUBTRACT = [
+        (fw1, 5, FermiSentence({fw1: -1, fw4: 5})),  # int
+        (fw2, 2.8, FermiSentence({fw2: -1, fw4: 2.8})),  # float
+        (fw3, (1 + 3j), FermiSentence({fw3: -1, fw4: (1 + 3j)})),  # complex
+        (fw1, np.array([5]), FermiSentence({fw1: -1, fw4: 5})),  # numpy array
+        (fw2, pnp.array([2.8]), FermiSentence({fw2: -1, fw4: 2.8})),  # pennylane numpy array
+        (fw4, 2, FermiSentence({fw4: 1})),  # FermiWord is Identity
+    ]
+
+    @pytest.mark.parametrize("w, c, res", CONSTANTS_AND_WORDS_SUBTRACT)
+    def test_subtract_fermi_words_from_constant(self, w, c, res):
+        """Test that subtracting a constant (int, float or complex) from a FermiWord
+        returns the expected FermiSentence"""
+        diff = c - w
+        # due to rounding, the actual result for floats is
+        # e.g. -0.19999999999999... instead of 0.2, so we round to compare
+        diff_rounded = FermiSentence(
+            {k: round(v, 2) if isinstance(v, float) else v for k, v in diff.items()}
```

Comment:
Test should round to more than two decimal places, maybe you can use np.allclose and iterate over both dictionaries to compare the terms?